In [384]:
import pandas as pd

# Read the CSV file
df = pd.read_csv("/home/pablo.canosa/wip/Mask2Former/output/tiempos.csv", header=None, names=["key", "value"])

# Calculate statistics for each key
stats = df.groupby("key")["value"].agg(['mean', 'median', 'std'])

# Display the statistics
print(stats)

                         mean    median       std
key                                              
backbone             0.013770  0.013153  0.009886
pixel_decoder        0.038965  0.038792  0.003981
transformer_decoder  0.036482  0.036827  0.002929


In [385]:
import pandas as pd

# Read the CSV file
df = pd.read_csv("/home/pablo.canosa/wip/models_and_results/TOP_42k_iters_rgb_results/output/times_2k_iters_r50_RGB.csv", header=None, names=["key", "value"])

# Calculate statistics for each key
stats = df.groupby("key")["value"].agg(['mean', 'median', 'std'])

# Display the statistics
print(stats)

                         mean    median       std
key                                              
backbone             0.014388  0.014241  0.009291
pixel_decoder        0.035918  0.035450  0.004083
transformer_decoder  0.035522  0.035537  0.003951


In [386]:
import pandas as pd

# Read the CSV file
df = pd.read_csv("/home/pablo.canosa/wip/models_and_results/TOP_57k_iters_NIRRGB_results/output/times_2k_iters_r50_NIRRGB.csv", header=None, names=["key", "value"])

# Calculate statistics for each key
stats = df.groupby("key")["value"].agg(['mean', 'median', 'std'])

# Display the statistics
print(stats)

                         mean    median       std
key                                              
backbone             0.011843  0.011172  0.008522
pixel_decoder        0.035329  0.035536  0.002807
transformer_decoder  0.032875  0.031272  0.002620


In [387]:
import pandas as pd

# Read the CSV file
df = pd.read_csv("/home/pablo.canosa/wip/models_and_results/RESNET100-RGB/output/times_2k_iters_r100_RGB.csv", header=None, names=["key", "value"])

# Calculate statistics for each key
stats = df.groupby("key")["value"].agg(['mean', 'median', 'std'])

# Display the statistics
print(stats)

                         mean    median       std
key                                              
backbone             0.023213  0.023560  0.010107
pixel_decoder        0.035042  0.034442  0.004274
transformer_decoder  0.033574  0.032506  0.003151


In [388]:
import pandas as pd

# Read the CSV file
df = pd.read_csv("/home/pablo.canosa/wip/models_and_results/RESNET100-NIRRGB/output/times_2k_iters_r100_NIRRGB.csv", header=None, names=["key", "value"])

# Calculate statistics for each key
stats = df.groupby("key")["value"].agg(['mean', 'median', 'std'])

# Display the statistics
print(stats)

                         mean    median       std
key                                              
backbone             0.022216  0.023019  0.009021
pixel_decoder        0.035335  0.034866  0.002953
transformer_decoder  0.033091  0.032220  0.002552


# Comparación de tiempos GPU 5

## Multispectral R50

In [14]:
import pandas as pd

# Read the CSV file
df = pd.read_csv("/home/pablo.canosa/Desktop/resultados_tiempos_GPU5/multi-paralelo-r50/tiempos.csv", header=None, names=["key", "value"])

# Calculate statistics for each key
stats = df.groupby("key")["value"].agg(['mean', 'median', 'std'])

# Display the statistics
print("PARALELO Multiespectral R50")
print(stats)

PARALELO Multiespectral R50
                         mean    median       std
key                                              
backbone             0.031572  0.026499  0.011854
backwards            0.198461  0.181365  0.063906
pixel_decoder        0.074735  0.068221  0.032904
transformer_decoder  0.076697  0.073036  0.021134


In [23]:
import pandas as pd

# Read the CSV file
df = pd.read_csv("/home/pablo.canosa/Desktop/resultados_tiempos_GPU5/multi-secuencial-r50/tiempos.csv", header=None, names=["key", "value"])

# Calculate statistics for each key
stats = df.groupby("key")["value"].agg(['mean', 'median', 'std'])

# Display the statistics
print("SECUENCIAL Multiespectral R50")
print(stats)

SECUENCIAL Multiespectral R50
                         mean    median       std
key                                              
backbone             0.027791  0.026232  0.011777
backwards            0.162018  0.147231  0.048211
pixel_decoder        0.193353  0.193865  0.070671
transformer_decoder  0.127982  0.135697  0.037667


## RGB R50

In [16]:
import pandas as pd

# Read the CSV file
df = pd.read_csv("/home/pablo.canosa/Desktop/resultados_tiempos_GPU5/RGB-Paralelo-r50/tiempos.csv", header=None, names=["key", "value"])

# Calculate statistics for each key
stats = df.groupby("key")["value"].agg(['mean', 'median', 'std'])

# Display the statistics
print("PARALELO RGB R50")
print(stats)

PARALELO RGB R50
                         mean    median       std
key                                              
backbone             0.030694  0.026156  0.011523
backwards            0.201217  0.184286  0.062160
pixel_decoder        0.075044  0.066762  0.028987
transformer_decoder  0.078012  0.072950  0.019734


In [17]:
import pandas as pd

# Read the CSV file
df = pd.read_csv("/home/pablo.canosa/Desktop/resultados_tiempos_GPU5/RGB-Secuencial-r50/tiempos.csv", header=None, names=["key", "value"])

# Calculate statistics for each key
stats = df.groupby("key")["value"].agg(['mean', 'median', 'std'])

# Display the statistics
print("SECUENCIAL RGB R50")
print(stats)

SECUENCIAL RGB R50
                         mean    median       std
key                                              
backbone             0.029099  0.026580  0.008520
backwards            0.162493  0.147035  0.048167
pixel_decoder        0.185367  0.170368  0.069592
transformer_decoder  0.128376  0.135059  0.034299


## Multispectral R100

In [18]:
import pandas as pd

# Read the CSV file
df = pd.read_csv("/home/pablo.canosa/Desktop/resultados_tiempos_GPU5/multi-parelelo-r100/tiempos.csv", header=None, names=["key", "value"])

# Calculate statistics for each key
stats = df.groupby("key")["value"].agg(['mean', 'median', 'std'])

# Display the statistics
print("PARALELO Multiespectral R100")
print(stats)

PARALELO Multiespectral R100
                         mean    median       std
key                                              
backbone             0.055030  0.048501  0.016745
backwards            0.284712  0.253624  0.095027
pixel_decoder        0.079090  0.067975  0.037734
transformer_decoder  0.074729  0.070365  0.020244


In [20]:
import pandas as pd

# Read the CSV file
df = pd.read_csv("/home/pablo.canosa/Desktop/resultados_tiempos_GPU5/multi-secuencial-r100/tiempos.csv", header=None, names=["key", "value"])

# Calculate statistics for each key
stats = df.groupby("key")["value"].agg(['mean', 'median', 'std'])

# Display the statistics
print("SECUENCIAL Multiespectral R100")
print(stats)

SECUENCIAL Multiespectral R100
                         mean    median       std
key                                              
backbone             0.065570  0.065478  0.018684
backwards            0.416567  0.400780  0.123027
pixel_decoder        0.209508  0.213749  0.071581
transformer_decoder  0.129325  0.138081  0.039357


## RGB R100

In [21]:
import pandas as pd

# Read the CSV file
df = pd.read_csv("/home/pablo.canosa/Desktop/resultados_tiempos_GPU5/RGB-Paralelo-r100/tiempos.csv", header=None, names=["key", "value"])

# Calculate statistics for each key
stats = df.groupby("key")["value"].agg(['mean', 'median', 'std'])

# Display the statistics
print("PARALELO RGB R100")
print(stats)

PARALELO RGB R100
                         mean    median       std
key                                              
backbone             0.060474  0.051229  0.020912
backwards            0.289733  0.260600  0.096053
pixel_decoder        0.077924  0.066465  0.035986
transformer_decoder  0.077770  0.073394  0.022053


In [22]:
import pandas as pd

# Read the CSV file
df = pd.read_csv("/home/pablo.canosa/Desktop/resultados_tiempos_GPU5/RGB-Secuencial-r100/tiempos.csv", header=None, names=["key", "value"])

# Calculate statistics for each key
stats = df.groupby("key")["value"].agg(['mean', 'median', 'std'])

# Display the statistics
print("SECUENCIAL RGB R100")
print(stats)

SECUENCIAL RGB R100
                         mean    median       std
key                                              
backbone             0.068291  0.075077  0.018414
backwards            0.412558  0.393264  0.118516
pixel_decoder        0.212931  0.230972  0.072766
transformer_decoder  0.132718  0.141124  0.037358


Here is the updated table with the rows ordered as **Backbone**, **Pixel Decoder**, **Transformer Decoder**, and **Backwards**:  

| Key                   | PARALELO Multispectral R50 | SECUENCIAL Multispectral R50 | PARALELO RGB R50 | SECUENCIAL RGB R50 | PARALELO Multispectral R100 | SECUENCIAL Multispectral R100 | PARALELO RGB R100 | SECUENCIAL RGB R100 |
|-----------------------|----------------------------|------------------------------|------------------|-------------------|-----------------------------|-------------------------------|-------------------|---------------------|
| **Backbone**          | 0.031572                  | 0.027791                    | 0.030694         | 0.029099          | 0.055030                   | 0.065570                     | 0.060474          | 0.068291            |
| **Pixel Decoder**     | 0.074735                  | 0.193353                    | 0.075044         | 0.185367          | 0.079090                   | 0.209508                     | 0.077924          | 0.212931            |
| **Transformer Decoder**| 0.076697                  | 0.127982                    | 0.078012         | 0.128376          | 0.074729                   | 0.129325                     | 0.077770          | 0.132718            |
| **Backwards**         | 0.198461                  | 0.162018                    | 0.201217         | 0.162493          | 0.284712                   | 0.416567                     | 0.289733          | 0.412558            |